### PR TITLE
refactor guides and add ability to edit them

### PR DESCRIPTION
**Pull request by amosmim**
changes in high level:
+ switch the use of GuideInfo with UserInfo
+ switch the use of Guides DB with Users DB
+ had ability to edit them from guides tabs in admins screen

### CODE REVIEW

**Comment by amosmim** [comment]
I add to this pull request the following features:

- change "guides" tab to "user mangement"
- support delete user info
- mange admin and user-type fields in DB
- some more bugs
- mark that user is guide with icon

@ariel-bentu, as far as I remember, this is all we speak about.
Please let me know if I miss something.

File: src/api.tsx [hunk 1]
```diff
@@ -427,36 +416,117 @@ export async function addMedia(name: string, type: "icon" | "photo", file: File)
...
+export function isUserAdmin(user: UserInfo) : Promise<boolean> {
+    if (!user._ref) 
+        return new Promise(() => false);
+
+    const docRef = doc(db, Collections.USERS_COLLECTION, user._ref.id, Collections.USER_SYSTEM_SUBCOLLECTION, "Default");
+    return getDoc(docRef).then(systemDoc => {return (systemDoc.exists() && systemDoc.data().admin)});
+}
+
+function UpdateUserAdminState(_ref: DocumentReference, isAdmin : boolean) {
+    const docRef = doc(db, Collections.USERS_COLLECTION, _ref.id, Collections.USER_SYSTEM_SUBCOLLECTION, "Default");
```
Comment (ariel-bentu) [inline src/api.tsx:438]: I would add protection so one user would not revoke his own admin rights

File: src/api.tsx [hunk 2]
```diff
@@ -427,36 +416,117 @@ export async function addMedia(name: string, type: "icon" | "photo", file: File)
...
     const storage = getStorage(app);
     const storageRef = ref(storage);
     const mediaRef = ref(storageRef, 'media');
     const folderRef = ref(mediaRef, 'guides_pics');
     const resourceRef = ref(folderRef, pic.name);
 
+    return resourceRef;
+}
+
+export function isUserAdmin(user: UserInfo) : Promise<boolean> {
```
Comment (ariel-bentu) [inline src/api.tsx:429]: simpler code: define the function async and return boolean:
```
export async function isUserAdmin(u:UserInfo): boolean {
if (!user._ref) {
return false;
...

File: src/api.tsx [hunk 3]
```diff
@@ -427,36 +416,117 @@ export async function addMedia(name: string, type: "icon" | "photo", file: File)
...
+
+function UpdateUserAdminState(_ref: DocumentReference, isAdmin : boolean) {
+    const docRef = doc(db, Collections.USERS_COLLECTION, _ref.id, Collections.USER_SYSTEM_SUBCOLLECTION, "Default");
+    return setDoc(docRef, {admin : isAdmin});
+}
+
+export async function editUserInfo(_ref: DocumentReference, pic: File | null, userInfo : UserInfo, isAdmin : boolean) {
+    console.log("we got ref need to update " + userInfo.fname + " " + userInfo.lname +" , " + (pic ? pic.name: "NULL") + " , " + _ref.id);
+
+    return getDoc(_ref).then((g)=> {
```
Comment (ariel-bentu) [inline src/api.tsx:445]: use exists

File: src/edit-guide-info.tsx [hunk 4]
```diff
@@ -1,41 +1,106 @@
...
+    const [preview, setPreview] = useState<string|undefined>(guide_info.avatar?.url);
+    const [fname, setFName] = useState<string>(guide_info.fname);
+    const [lname, setLName] = useState<string>(guide_info.lname);
+    const [email, setEmail] = useState<string>(guide_info._ref ? guide_info._ref.id : "");
+    const [admin, setAdmin] = useState<boolean>(false);
+    const [read_DB_once, setReadDBOnce] =  useState<boolean>(false);
+    const [type, setType] = useState<UserType>(guide_info.type);
+    const [adminInDB, setadminInDB] =  useState<boolean>(false);
 
+    if (!read_DB_once) {
```
Comment (ariel-bentu) [inline src/edit-guide-info.tsx:20]: For this exactly they invented the useEffect hook

**Review by ariel-bentu — APPROVED** [review]
Hi,
Some comments, but all in all - nice work, ready to merge :)

File: src/api.tsx (hunk 2 above)
Comment (amosmim) [reply src/api.tsx:429]: Fix in main branch

**Pull request merged**